// angular2-notifications/src/simple-notifications/components/notification.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '../pipes/max.pipe';
import * as import3 from './notification.component';
import * as import4 from '../services/notifications.service';
import * as import5 from '@angular/platform-browser';
const styles_NotificationComponent:any[] = ['\n        .simple-notification {\n            width: 100%;\n            padding: 10px 20px;\n            box-sizing: border-box;\n            position: relative;\n            float: left;\n            margin-bottom: 10px;\n            color: #fff;\n            cursor: pointer;\n            transition: all 0.5s;\n        }\n\n        .simple-notification .sn-title {\n            margin: 0;\n            padding: 0 50px 0 0;\n            line-height: 30px;\n            font-size: 20px;\n        }\n\n        .simple-notification .sn-content {\n            margin: 0;\n            font-size: 16px;\n            padding: 0 50px 0 0;\n            line-height: 20px;\n        }\n\n        .simple-notification .icon {\n            position: absolute;\n            box-sizing: border-box;\n            top: 0;\n            right: 0;\n            width: 70px;\n            height: 70px;\n            padding: 10px;\n        }\n\n        .simple-notification .icon svg {\n            fill: #fff;\n            width: 100%;\n            height: 100%;\n        }\n\n        .simple-notification .icon svg g {\n            fill: #fff;\n        }\n\n        .simple-notification.rtl-mode {\n            direction: rtl;\n        }\n\n        .simple-notification.rtl-mode .sn-content {\n            padding: 0 0 0 50px;\n        }\n\n        .simple-notification.rtl-mode svg {\n            left: 0;\n            right: auto;\n        }\n\n        .simple-notification.error { background: #F44336; }\n        .simple-notification.success { background: #8BC34A; }\n        .simple-notification.alert { background: #ffdb5b; }\n        .simple-notification.info { background: #03A9F4; }\n        .simple-notification.warn { background: #ffdb5b; }\n\n        .simple-notification .sn-progress-loader {\n            position: absolute;\n            top: 0;\n            left: 0;\n            width: 100%;\n            height: 5px;\n        }\n\n        .simple-notification .sn-progress-loader span {\n            float: left;\n            height: 100%;\n        }\n\n        .simple-notification.success .sn-progress-loader span { background: #689F38; }\n        .simple-notification.error .sn-progress-loader span { background: #D32F2F; }\n        .simple-notification.alert .sn-progress-loader span { background: #edc242; }\n        .simple-notification.info .sn-progress-loader span { background: #0288D1; }\n        .simple-notification.warn .sn-progress-loader span { background: #edc242; }\n        .simple-notification.bare .sn-progress-loader span { background: #ccc; }\n\n        .simple-notification.warn div .sn-title { color: #444; }\n        .simple-notification.warn div .sn-content { color: #444; }\n    '];
export const RenderType_NotificationComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NotificationComponent,
      data: {'animation': [{
        name: 'enterLeave',
        definitions: [
          {
            type: 0,
            name: 'fromRight',
            styles: {
              type: 6,
              styles: {
                opacity: 1,
                transform: 'translateX(0)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: '* => fromRight',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'translateX(5%)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '400ms ease-in-out'
              }

            ]

          }
          ,
          {
            type: 0,
            name: 'fromRightOut',
            styles: {
              type: 6,
              styles: {
                opacity: 0,
                transform: 'translateX(-5%)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: 'fromRight => fromRightOut',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 1,
                  transform: 'translateX(0)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '300ms ease-in-out'
              }

            ]

          }
          ,
          {
            type: 0,
            name: 'fromLeft',
            styles: {
              type: 6,
              styles: {
                opacity: 1,
                transform: 'translateX(0)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: '* => fromLeft',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'translateX(-5%)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '400ms ease-in-out'
              }

            ]

          }
          ,
          {
            type: 0,
            name: 'fromLeftOut',
            styles: {
              type: 6,
              styles: {
                opacity: 0,
                transform: 'translateX(5%)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: 'fromLeft => fromLeftOut',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 1,
                  transform: 'translateX(0)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '300ms ease-in-out'
              }

            ]

          }
          ,
          {
            type: 0,
            name: 'scale',
            styles: {
              type: 6,
              styles: {
                opacity: 1,
                transform: 'scale(1)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: '* => scale',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'scale(0)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '400ms ease-in-out'
              }

            ]

          }
          ,
          {
            type: 0,
            name: 'scaleOut',
            styles: {
              type: 6,
              styles: {
                opacity: 0,
                transform: 'scale(0)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: 'scale => scaleOut',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 1,
                  transform: 'scale(1)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '400ms ease-in-out'
              }

            ]

          }
          ,
          {
            type: 0,
            name: 'rotate',
            styles: {
              type: 6,
              styles: {
                opacity: 1,
                transform: 'rotate(0deg)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: '* => rotate',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  transform: 'rotate(5deg)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '400ms ease-in-out'
              }

            ]

          }
          ,
          {
            type: 0,
            name: 'rotateOut',
            styles: {
              type: 6,
              styles: {
                opacity: 0,
                transform: 'rotate(-5deg)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: 'rotate => rotateOut',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 1,
                  transform: 'rotate(0deg)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '400ms ease-in-out'
              }

            ]

          }

        ]

      }
  ]}
}
);
function View_NotificationComponent_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'icon'
      ]
      ],[[
        8,
        'innerHTML',
        1
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.safeSvg;
    ck(v,0,0,currVal_0);
  });
}
function View_NotificationComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),11,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'sn-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'sn-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    import0.ɵppd(2),
    (l()(),import0.ɵted((null as any),['\n\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NotificationComponent_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = (co.item.icon !== 'bare');
    ck(v,10,0,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.item.title;
    ck(v,3,0,currVal_0);
    const currVal_1:any = import0.ɵunv(v,6,0,ck(v,7,0,import0.ɵnov((<any>v.parent),0),co.item.content,co.maxLength));
    ck(v,6,0,currVal_1);
  });
}
function View_NotificationComponent_3(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',([] as any[]),[[
        8,
        'innerHTML',
        1
      ]
  ],(null as any),(null as any),(null as any),(null as any)))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.item.html;
    ck(v,0,0,currVal_0);
  });
}
function View_NotificationComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[[
        'class',
        'sn-progress-loader'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['width']),
    (l()(),import0.ɵted((null as any),['\n            ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,4,0,(co.progressWidth + '%'));
    ck(v,3,0,currVal_0);
  },(null as any));
}
export function View_NotificationComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵpid(0,import2.MaxPipe,([] as any[])),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),12,'div',[[
        'class',
        'simple-notification'
      ]
      ],[[
        24,
        '@enterLeave',
        0
      ]
    ],[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'mouseenter'
      ]
      ,
      [
        (null as any),
        'mouseleave'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:import3.NotificationComponent = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.onClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('mouseenter' === en)) {
        const pd_1:any = ((<any>co.onEnter()) !== false);
        ad = (pd_1 && ad);
      }
      if (('mouseleave' === en)) {
        const pd_2:any = ((<any>co.onLeave()) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'alert',
      'error',
      'warn',
      'success',
      'info',
      'bare',
      'rtl-mode'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NotificationComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NotificationComponent_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n            '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NotificationComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import3.NotificationComponent = v.component;
    const currVal_1:any = co.theClass;
    const currVal_2:any = ck(v,4,0,(co.item.type === 'alert'),(co.item.type === 'error'),(co.item.type === 'warn'),(co.item.type === 'success'),(co.item.type === 'info'),(co.item.type === 'bare'),co.rtl);
    ck(v,3,0,currVal_1,currVal_2);
    const currVal_3:boolean = !co.item.html;
    ck(v,7,0,currVal_3);
    const currVal_4:any = co.item.html;
    ck(v,10,0,currVal_4);
    const currVal_5:any = co.showProgressBar;
    ck(v,13,0,currVal_5);
  },(ck,v) => {
    var co:import3.NotificationComponent = v.component;
    const currVal_0:any = co.item.state;
    ck(v,2,0,currVal_0);
  });
}
function View_NotificationComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'simple-notification',([] as any[]),(null as any),(null as any),(null as any),View_NotificationComponent_0,RenderType_NotificationComponent)),
    import0.ɵdid(122880,(null as any),0,import3.NotificationComponent,[
      import4.NotificationsService,
      import5.DomSanitizer,
      import0.NgZone
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(null as any));
}
export const NotificationComponentNgFactory:import0.ComponentFactory<import3.NotificationComponent> = import0.ɵccf('simple-notification',import3.NotificationComponent,View_NotificationComponent_Host_0,{
  timeOut: 'timeOut',
  showProgressBar: 'showProgressBar',
  pauseOnHover: 'pauseOnHover',
  clickToClose: 'clickToClose',
  maxLength: 'maxLength',
  theClass: 'theClass',
  rtl: 'rtl',
  animate: 'animate',
  position: 'position',
  item: 'item'
}
,{},([] as any[]));
